src/Illuminate/Collections/Collection.php

```diff
@@ -1134,7 +1134,11 @@ public function shift($count = 1)
             throw new InvalidArgumentException('Number of shifted items may not be less than zero.');
         }
 
-        if ($count === 0 || $this->isEmpty()) {
+        if ($this->isEmpty()) {
```

Comment:
I think this is the correct placement for the `isEmpty()` check. We want to make sure that there are items to use before we continue to handle the rest.

Also this maintains the behaviour as it was prior to the mentioned PR.